itest/rust/src/framework/runner.rs

```diff
@@ -72,18 +89,71 @@ impl IntegrationTests {
         }
 
         let clock = Instant::now();
-        self.run_rust_tests(rust_tests, scene_tree, property_tests.clone());
+        self.run_rust_tests(rust_tests, scene_tree.clone(), property_tests.clone());
         let rust_time = clock.elapsed();
-        property_tests.free();
 
         let gdscript_time = if !focus_run {
             let extra_duration = self.run_gdscript_tests(gdscript_tests);
-            Some((clock.elapsed() - rust_time) + extra_duration)
+            Some((clock.elapsed() - rust_time, extra_duration))
         } else {
             None
         };
 
-        self.conclude_tests(rust_time, gdscript_time, allow_focus)
+        #[cfg(before_api = "4.2")]
+        {
+            use godot::obj::WithBaseField;
+
+            property_tests.free();
+
+            let result = Self::conclude_tests(
+                &self.stats,
+                rust_time,
+                gdscript_time.map(|(elapsed, extra)| elapsed + extra),
+                allow_focus,
+            );
+
+            // on_finished will call back into self, so we have to make self re-entrant. We also can't call on_finished in deferred mode,
+            // since it's not available under the 4.1 API.
+            let base = self.base_mut();
+            on_finished.callv(&[result.to_variant()].to_godot());
```

Comment:
```suggestion
            on_finished.callv(varray![&result]);
```
(not 100% sure if `&` is needed)